### DISCUSSION TITLE
Encode on the client and decode on the server without knowing the original dataset

### DISCUSSION THREAD

**Discussion by josued8a** [ananthakumaran/u#5] [Q&A]
I notice that your encoder and decoder need to know the original dataset, but I want to accomplish something really simple.

From the client side, a user will type a really long text, something around 2k characters max, there are no special characters or funny things, just regular text.

I want to send the request encrypted and compressed to a URL webpage, for the reason I need to do it that way.

On the server side, I don't know what they will send, I don't have the V1 dataset as your example, so I will try to do something like this:

/* Client-side:*/
var data = "{value='REALLY LONG TEXT WITH NO SPECIAL CHARACTERS, JUST TEXT, NUMBERS AND SOME CHARACTERS LIKE / - *'}";
var data_encoded = encode(data);

Request.Redirect("mydomain.com/webpage.aspx?data=data_encoded");
/*  Server-side: */
var data_encoded = Request.QueryString("data");
var data_decoded = decode(data_encoded);
.....

Do you think that I can accomplish this with your library?

**Comment by ananthakumaran** [comment]
This library only encodes the information in a form suitable for URL, not compress it. If you have just a plain string, there is not much it can do. You can use the varchar type, but that will just put the entire string as it is. This library helps in cases where you want to store user preferences/search filters etc where the set of options is limited.
